sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -117,6 +126,34 @@ ModuleInfo generateModuleInfo(Set<? extends TypeElement> annotations, RoundEnvir
                     .formatted(qName));
           }
 
+          Optional<ConstructorInfo> constructorInfo = Optional.empty();
+          if (mainObject) {
+            Optional<? extends Element> constructorDef =
+                typeElement.getEnclosedElements().stream()
+                    .filter(elt -> elt.getKind() == ElementKind.CONSTRUCTOR)
+                    .filter(elt -> !((ExecutableElement) elt).getParameters().isEmpty())
+                    .findFirst();
```

Comment:
Maybe we should throw an exception when there is more than one constructor

Reply:
Yeah, I didn't really know. I was wondering if we should add a specific annotation to specify which one is the right one.

Reply:
I've changed my mind and went on your direction. I know throw an exception if there's more than 1 non empty constructor. That way we ensure the behavior is the one we expect.